data/src/main/java/com/microsoft/azure/kusto/data/auth/ConnectionStringBuilder.java

```diff
@@ -58,6 +53,16 @@ private ConnectionStringBuilder(String clusterUrl) {
         this.userNameForTracing = null;
         this.clientVersionForTracing = null;
         this.applicationNameForTracing = null;
+        this.initProcessNameForTracing();
+    }
+
+    private void initProcessNameForTracing() {
+        this.processNameForTracing = System.getProperty("sun.java.command");
+        this.userNameForTracing = System.getProperty("user.name");
+        if (processNameForTracing != null) {
+            String[] p = processNameForTracing.split(" ")[0].split("\\.");
```

Comment:
Comment explaining what's going on here.
The '\\.' is suspicious, it looks windows specific, and path specific?

Plus, make the code more defensive.
You are doing a lot of checks that can throw, and we don't want to not to be able to create a KSCB if the format of the property isn't correct, or it doesn't exist.

Reply:
What throws here

Reply:
split always returns at least one element